src/logohunter.py

```diff
@@ -0,0 +1,349 @@
+        elif os.path.isfile(FLAGS.input_images):
+            FLAGS.input_images = [ os.path.abspath(FLAGS.input_images)  ]
+        else:
+            exit('Error: path not found: {}'.format(FLAGS.input_images))
+
+
+        print('Found {} input brands: {}...'.format(len(FLAGS.input_brands), [ os.path.basename(f) for f in FLAGS.input_brands[:5]]))
+        print('Found {} input images: {}...'.format(len(FLAGS.input_images), [ os.path.basename(f) for f in FLAGS.input_images[:5]]))
+
+        output_path = FLAGS.output
+        if not os.path.exists(output_path):
+            os.makedirs(output_path)
+
+        # define YOLO logo detector
+        yolo = YOLO(**{"model_path": FLAGS.model_path,
+                    "anchors_path": FLAGS.anchors_path,
+                    "classes_path": FLAGS.classes_path,
+                    "score" : FLAGS.score,
+                    "gpu_num" : FLAGS.gpu_num,
+                    "model_image_size" : (416, 416),
+                    }
+                   )
+
+
+        input_paths = sorted(FLAGS.input_brands)
+        input_labels = [ os.path.basename(s).split('test_')[-1].split('.')[0] for s in input_paths]
+
+        ## load pre-processed features database
+        brand_map, features = load_features(FLAGS.features)
+
+        ## load inception model
+        model, preprocess_input = load_extractor_model()
+        my_preprocess = lambda x: chain_preprocess(x, input_shape, preprocess_input)
+
+        # compute cosine similarity between input brand images and all LogosInTheWild logos
+        ( img_input, feat_input, sim_cutoff, (bins, cdf_list)
+        ) = load_brands_compute_cutoffs(input_paths, (model, my_preprocess), features, sim_threshold)
+
+        start = timer()
+        # cycle trough input images, look for logos and then match them against inputs
+        text_out = ''
+        for i, img_path in enumerate(FLAGS.input_images):
+            prediction, r_image = detect_logo(yolo, img_path, save_img = save_img_logo,
+                                              save_img_path = FLAGS.output,
+                                              postfix='_logo')
+
+            text = match_logo(img_path, prediction, (model, my_preprocess),
+                    (feat_input, sim_cutoff, bins, cdf_list, input_labels),
+                    save_img = save_img_match, save_img_path=FLAGS.output)
+            print(text)
+            text_out += text
+
+        if save_to_txt:
+            with open(output_txt,'w') as txtfile:
+                txtfile.write(text_out)
+
+        end = timer()
+        print('Processed {} images in {:.1f}sec - {:.1f}FPS'.format(
+             len(FLAGS.input_images), end-start, len(FLAGS.input_images)/(end-start)
+             ))
+
+    # video mode
+    # elif FLAGS.video:
+    else:
+        print("Must specify either --image or --video.  See usage with --help.")
```

Comment:
overall the main function is too long and does three things (Image detection mode, Test routine, detection + similarity mode). I would suggest to break this into 3 scripts, so it is easy to follow and use, you could reuse the functions

Reply:
Thanks, will do!